lua/nvim-lsp-installer/servers/arduino_language_server/init.lua

```diff
@@ -0,0 +1,75 @@
+local path = require "nvim-lsp-installer.path"
+local std = require "nvim-lsp-installer.installers.std"
+local installers = require "nvim-lsp-installer.installers"
+local server = require "nvim-lsp-installer.server"
+local go = require "nvim-lsp-installer.installers.go"
+local shell = require "nvim-lsp-installer.installers.shell"
+local platform = require "nvim-lsp-installer.platform"
+local context = require "nvim-lsp-installer.installers.context"
+local Data = require "nvim-lsp-installer.data"
+
+local coalesce, when = Data.coalesce, Data.when
+
+return function(name, root_dir)
+    local arduino_cli_installer = installers.pipe {
+        context.use_github_release_file("arduino/arduino-cli", function(version)
+            return coalesce(
+                when(
+                    platform.is_mac,
+                    coalesce(
+                        when(platform.arch == "x64", "arduino-cli_%s_macOS_64bit.tar.gz")
+                    )
+                ),
+                when(
+                    platform.is_linux,
+                    coalesce(
+                        when(platform.arch == "x64", "arduino-cli_%s_Linux_64bit.tar.gz"),
+                        when(platform.arch == "x86", "arduino-cli_%s_Linux_32bit.tar.gz"),
+                        when(platform.arch == "arm64", "arduino-cli_%s_Linux_ARM64.tar.gz"),
+                        when(platform.arch == "armv6", "arduino-cli_%s_Linux_ARMv6.tar.gz"),
+                        when(platform.arch == "armv7", "arduino-cli_%s_Linux_ARMv7.tar.gz")
+                    )
+                ),
+                when(
+                    platform.is_win,
+                    coalesce(
+                        when(platform.arch == "x64", "arduino-cli_%s_Windows_64bit.zip"),
+                        when(platform.arch == "x86", "arduino-cli_%s_Windows_32bit.zip")
+                    )
+                )
+            ):format(version)
```

Comment:
Calling `:format` on this expression might cause nil pointer errors, for example if you're on mac but on arm64. This function can return `nil`, in which case the installer will interpret it as an error caused by the current system not being supported. What about:

```lua
local target_file = coalesce(...)
return target_file and target_file:format(version)
```